src/pretix/base/services/orders.py

```diff
@@ -1289,9 +1289,14 @@ def send_expiry_warnings(sender, **kwargs):
     event_id = None
 
     for o in Order.objects.filter(
-        expires__gte=today, expiry_reminder_sent=False, status=Order.STATUS_PENDING,
-        datetime__lte=now() - timedelta(hours=2), require_approval=False
+            expires__gte=today, expiry_reminder_sent=False, status=Order.STATUS_PENDING,
+            datetime__lte=now() - timedelta(hours=2), require_approval=False
     ).only('pk', 'event_id', 'expires').order_by('event_id'):
+
+        lp = o.payments.last()
+        if lp and lp.payment_provider.prevent_reminder_mail(o, lp):
+            continue
```

Comment:
I don't think we should prevent the reminder email if the last payment is e.g. in status `failed`/`canceled` (the user will need to do something to complete the payment) or in status `completed` (the amount of payment is not sufficient and the user needs to do another payment). I think this should only respect payments in status `created` and `pending`.

Reply:
That makes sense. As a general condition that would better be handled here directly and not in the .prevent_reminder_mail function so that other payment providers could potentially say 'nope, don't care', right?

Reply:
I was thinking about this as well, and I think it's better handled here. A payment provider could *still* implement their own distinction between created and pending, but we should exclude the "clear-cut cases" here in core

Reply:
I changed it accordingly, but I want to look at the tests again, because new ones for this didn't work correctly, so I changed the pr into a draft again, please don't merge early